app/controllers/two_factor_authentication/personal_key_verification_controller.rb

```diff
@@ -81,8 +81,10 @@ def handle_valid_otp
       handle_valid_otp_for_authentication_context
       if decorated_user.identity_verified? || decorated_user.password_reset_profile.present?
         redirect_to manage_personal_key_url
+      elsif MfaPolicy.new(current_user, user_session[:signing_up]).sufficient_factors_enabled?
+        redirect_to after_multiple_2fa_sign_up
       else
-        redirect_to two_2fa_setup
+        redirect_to two_factor_options_url
```

Comment:
Not sure if this was the cleanest way to do this so suggestions welcome! 
The elsif/else statements are basically the same logic as `two_2fa_setup` in `application_controller.rb`, but I copied it over and changed the `else` to redirect to `two_factor_options_url`, because you don't want to see the `two_factor_options_success` screen when going through the personal key flow